com.unity.render-pipelines.high-definition/Runtime/Lighting/Light/HDAdditionalLightData.cs

```diff
@@ -1296,6 +1355,8 @@ public float shadowDimmer
                     return;
 
                 m_ShadowDimmer = Mathf.Clamp01(value);
+                if (lightEntity.valid)
```

Comment:
Putting here as a random place in code. Given how often we do the validity check, what are the situations in which the lightEntity is invalid, but the lighting data is perfectly valid? 
Can we enforce validity as early as possible and get rid of all the checks?

Reply:
So this is very very tricky.
A render entity gets created on OnEnable, and destroyed during OnDisable. This has carefully been done, otherwise would require figuring out when to delete the entity (cant be done at destruction since this doenst clear the buffers). 
Because of OnEanble/OnDisable, say during upgrade scripts, or someone modifying a disabled gameobject through a script could potentially have lightEntity being invalid. These functions are rarely hit though for many many lights. I recommend paying for the cost here for that reason.

Reply:
The functions might be actually be called a lot of times in user land as they are the way the user can drive light by script and so I expect multiple of them might be called per light. The overhead of the branch is not going to be massive (especially in the grand scope of things and our current state), but still it'd be nice to think of it at some point. 

I do see the point, but worth thinking about it if we ever reach a stage of needing a further push with a collection of micro-optimizations. 
I assume on Awake() and OnDestroy() with some checks on whether the light is enabled or not when populating the processed list could work? You say OnDestroy() wouldn't clear the buffer, what buffers? 

Regardless this is def. not a blocking point for me, just discussion :)

Reply:
The buffers I talk about are the NativeArray<> holding the entity datas.
These need to live outside the HDRP Pipeline lifetime. Because the pipeline gets instantiated and destroyed outside OnAwake / OnDestroy etc.
Thus, we Dispose the global NativeArray<> containing all the entity data when the last HDAdditionalLightObject gets destroyed.
Every object calls OnDestroy always, even if you kill the editor / reload scripts.